src/System.Windows.Forms.Primitives/src/Interop/OleAut32/Interop.VARIANT.cs

```diff
@@ -23,33 +27,30 @@ public sealed class VARIANT
 
             public IntPtr data2;
 
+            public VARENUM Type => (vt & VARENUM.TYPEMASK);
+
             public bool Byref => (vt & VARENUM.BYREF) != 0;
 
             public void Clear()
             {
-                if ((vt == VARENUM.UNKNOWN || vt == VARENUM.DISPATCH) && data1 != IntPtr.Zero)
-                {
-                    Marshal.Release(data1);
-                }
-
-                if (vt == VARENUM.BSTR && data1 != IntPtr.Zero)
+                fixed (VARIANT* pThis = &this)
                 {
-                    Oleaut32.SysFreeString(data1);
+                    VariantClear(pThis);
                 }
 
-                data1 = data2 = IntPtr.Zero;
                 vt = VARENUM.EMPTY;
+                data1 = IntPtr.Zero;
+                data1 = IntPtr.Zero;
```

Comment:
nit: redundant assignment
```suggestion
                data1 = IntPtr.Zero;
```